components/api/src/main/java/com/hotels/styx/api/HttpRequest.java

```diff
@@ -728,10 +728,10 @@ private void ensureContentLengthIsValid() {
             return headerValues.isEmpty() ? Optional.empty() : Optional.of(headerValues.get(0));
         }
 
-        private static boolean isInteger(String contentLength) {
+        private static boolean isPositiveInteger(String contentLength) {
```

Comment:
```suggestion
        private static boolean isNonNegativeInteger(String contentLength) {
```
Generally the term "positive integer" refers to an integer greater than zero. I'm not sure if there is an official ruling for numbers in Java, but this is the use of the term I am familiar with.